decorator.rb

```diff
@@ -0,0 +1,28 @@
+require_relative './nameable'
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    word_output = @nameable.correct_name
+    if word_output.length <= 10
+      word_output
+    else
+      word_output[0..10]
+    end
```

Comment:
- [ ] This method cuts off the names at 11 characters instead of 10. You can correct this by using [0..9], instead of [0..10]

  ![dfg](https://user-images.githubusercontent.com/81468675/192769017-8dd5128f-151d-4239-880a-d8ebb1187efb.jpg)

- [ ] **[OPTIONAL]** I think we can make this more concise by doing this instead

  ```
  def correct_name
    @nameable.correct_name[0..9]
  end
  ```

Reply:
@julie-ify  Thank you for pointing this out, I have implemented the change.